pulp_ansible/app/models.py

```diff
@@ -178,6 +179,22 @@ class Meta:
         ]
 
 
+class CollectionVersionSignature(Content):
+    """
+    A content type representing a signature that is attached to a content unit
+    # This model might live in pulpcore if it can be abstracted.
+    """
+    TYPE = "collection_signature"
+
+    content = models.ForeignKey(CollectionVersion, on_delete=models.CASCADE, related_name="signatures")
+    public_key = models.TextField()  # How long can public keys be?
+    pubkey_fingerprint = models.TextField()
+    signing_service = models.ForeignKey(SigningService, on_delete=models.SET_NULL, related_name="signatures", null=True)
```

Comment:
If we have a FK to the signing_service do we need the `public_key` and `pubkey_fingerprint` ?

Reply:
This can be null, since I was assuming in the future when we sync signatures from a remote they wouldn't have an associated signing service, so you would at least need the `pubkey_fingerprint` for easy filtering.

Reply:
Yes, there is that case where we sync from remote and also the case for signing service being deleted or keys rotated but content not changed.

Signing Service must be detached from the signature, we could store as a soft reference.